commons/src/main/java/org/jetbrains/bsp/bazel/commons/BspCompileState.kt

```diff
@@ -0,0 +1,13 @@
+package org.jetbrains.bsp.bazel.commons;
+
+import ch.epfl.scala.bsp4j.TextDocumentIdentifier
+import java.util.concurrent.ConcurrentHashMap
+
+/**
+ * @property hasAnyProblems keeps track of problems in given file so BSP reporter
+ * can publish diagnostics with an empty array, to clear up former diagnostics.
+ * see: https://youtrack.jetbrains.com/issue/BAZEL-376
+ */
+class BspCompileState {
```

Comment:
Sure, I defined this class in a separate place just to write in Kotlin, but we can move this into `BazelBspCompilationManager`

Reply:
ohh shit github doesnt like me - there is one important comment missing;
 anyway - imo the state should be created and stored in `BazelBspCompilationManager` then just passed to `newBepServer` - it's the only place where it's used (and where it makes sense) and then the scope will be limited; also i want to get rid of the `commons/` or at least limit it's usage since there are kinda random things (things which actually could be in a better places)

Reply:
Makes sense, got rid of the `BazelBspCompilationState` and passing the `Map` (`ConcurrentHashMap`) object across places.
https://github.com/JetBrains/bazel-bsp/pull/381/commits/c43fdacf1dc4fbb5a0515c22c0c389b1034eafc7
and made a bit refactor later https://github.com/JetBrains/bazel-bsp/pull/381/commits/a361ded0e4f967b09efb4f355dd65697ccf726ef